### ISSUE TITLE
Confusing results w/ __filename and no-path-concat rule

### ISSUE THREAD

**Issue by pdehaan** [eslint/eslint#776]
### Steps to reproduce:

``` javascript
// only start the server if the file is called directly, otherwise wait until
// module.exports is called.
if (process.argv[1] === __filename) {
  module.exports();
}
```

My .eslintrc file doesnt explicitly mention the `no-path-concat` rule, but running `$ eslint **/*.js` gives me the following error:

``` bash
$ eslint **/*.js

scripts/run_locally.js
  16:9   error    ['PORT'] is better written in dot notation                      dot-notation
  27:6   error    Don't use process.exit(); throw an error instead                no-process-exit
  34:4   error    Use path.join() or path.resolve() instead of + to create paths  no-path-concat
  22:43  warning  signal is defined but never used                                no-unused-vars

✖ 4 problems
```
### Actual results:

> _Use `path.join()` or `path.resolve()` instead of `+` to create paths_
### Expected results:

No warning. I'm not using `path.join()`, `path.resolve()` or `+`. I simply referenced the `__filename` global.
### Workaround:

Explicitly specify the `no-path-concat` flag in your .eslintrc file, although this is a bit brute-force and undesirable:

``` json
{
  "env": {
    "node": true
  },
  "rules": {
    "no-path-concat": 0
  }
}
```
